### DISCUSSION TITLE
Minimum shell thickness vs Ensure vertical shell thickness

### DISCUSSION THREAD

**Discussion by cfytrok** [supermerill/SuperSlicer#3254] [Q&A]
What is the difference between these settings? It seems SuperSlicers "Minimum shell thicknes" replace PrusaSlicers "Ensure vertical shell thickness". But why it still visible in GUI?

**Comment by neophyl** [comment]
Not the same thing at all.  The ensure vertical shell thickness attempts to work out if any extra perimeters are needed on sloping areas.  So if it would print and not have enough existing perimeters below to support the perimeters above then it 'should' add extra to in effect fill in the gaps of too extreme a slope.  So it works on Perimeters.

Minimum shell thickness is essentially an override for the top and bottom solid layers.  So for example if you set 1mm in there and set 5 solid layers for top then on a 0.2 layer height profile you have 1mm and so does nothing.  However if you changed you layer height to say 0.1 then it would make the top layers 10 when slicing as it would need 10x 0.1 layers to meet your 1mm minimum constraint.  So it works on top/bottom layers.

I remember when it was added, iirc due to people changing settings like layer height or using the variable layer height function and ending up with really thin top layers because of that.

**Reply by cfytrok** [reply]
1. Tip in GUI says Ensure vertical shell thickness adds solid infill, not perimeters. Maybe you mixed up it with Extra perimeter setting?
2. Could you please send an example of how ensure vertical shell thickness works?
In my case it doesn't change gcode.
![image](https://user-images.githubusercontent.com/29511727/191936932-a9515381-bc09-4f1e-a1ef-bcc96b061e66.png)
[Shape-Sphere.zip](https://github.com/supermerill/SuperSlicer/files/9632656/Shape-Sphere.zip)
